app/controllers/v0/user_preferences_controller.rb

```diff
@@ -5,18 +5,37 @@ class UserPreferencesController < ApplicationController
     include Accountable
 
     def create
-      account  = current_user.account.presence || create_user_account
       response = SetUserPreferences.new(account, requested_user_preferences).execute!
 
       render json: response, serializer: UserPreferenceSerializer
     end
 
+    # This endpoint deletes all associated UserPreferences for a given Preference code.
+    # If the Preference doesn't exist, a 404 not_found will be returned.
+    #
+    # @param code - Required Preference code for UserPreferences to be deleted
+    #
+    def delete_all
+      destroy_user_preferences!
+      render json: {}, status: :ok, serializer: UserPreferenceSerializer
+    end
+
     def index
       render json: user_preferences, serializer: UserPreferenceSerializer
     end
 
     private
 
+    def account
+      @account ||= current_user.account.presence || create_user_account
+    end
+
+    def destroy_user_preferences!
+      preferences = UserPreference.for_preference_and_account(account.id, params[:code])
+      raise Common::Exceptions::RecordNotFound, params[:code] if preferences.empty?
+      preferences.destroy_all
```

Comment:
Perhaps we should validate for a successful deletion of all records?  Could be something like [this](https://github.com/department-of-veterans-affairs/vets-api/blob/master/app/services/user_preferences/grantor.rb#L105-L111)?